extension/intents/nicknames/nicknames.js

```diff
@@ -99,42 +99,18 @@ intentRunner.registerIntent({
 intentRunner.registerIntent({
   name: "nicknames.namePage",
   async run(context) {
-    const name = context.slots.name.toLowerCase();
+    const name = context.slots.name;
     const activeTab = await browserUtil.activeTab();
     const metadata = await pageMetadata.getMetadata(activeTab.id);
-    const result = await browser.storage.sync.get("pageNames");
-    const pageNames = result.pageNames || {};
-    pageNames[name] = metadata.url;
-    log.info("Added page name", name);
-    await browser.storage.sync.set({ pageNames });
+    intentRunner.registerPageName(name, metadata);
   },
 });
 
 intentRunner.registerIntent({
   name: "nicknames.removePageName",
   async run(context) {
-    const result = await browser.storage.sync.get("pageNames");
-    const pageNames = result.pageNames || {};
-    const getName = async () => {
-      const activeTab = await browserUtil.activeTab();
-      const metadata = await pageMetadata.getMetadata(activeTab.id);
-      return Object.keys(pageNames).find(
-        key => pageNames[key] === metadata.url
-      );
-    };
-    const name = context.slots.name || (await getName());
-    if (!name) {
-      const exc = new Error("No page name to remove");
-      exc.displayMessage = `This page does not have a name`;
-      throw exc;
-    }
-    if (!pageNames[name]) {
-      const exc = new Error("No page name to remove");
-      exc.displayMessage = `No page name "${name}" found`;
-      throw exc;
-    }
-    delete pageNames[name];
-    log.info("Removed page name", name);
-    await browser.storage.sync.set({ pageNames });
+    const name = context.slots.name;
+    await intentRunner.getRegisteredPageName(name);
+    intentRunner.registerPageName(name, null);
```

Comment:
Here's the error: this passes the second argument as `null`, but the registerPageName function expects an object with a property `url`. So it would work with `intentRunner.registerPageName(name, {url: null})`

Probably the best fix is to update `registerPageName` though so it can accept null as a second argument.

Reply:
Throws a `TypeError: can't access property "url", pagedata is null`

```
export async function registerPageName(name, pagedata) {
  name = name.toLowerCase();
  const creationDate = Date.now();
  const result = await browser.storage.sync.get("pageNames");
  pageNames = result.pageNames || {};
  const url = pagedata.url;
  pageNames[name] = url;

  if (!url) {
    delete pageNames[name];
    log.info("Removed nickname for page", name);
  } else {
    pageNames[name] = url;
    log.info("Added nickname for page", name, "->", url, creationDate);
  }
  await browser.storage.sync.set({ pageNames });
}
```

Reply:
But then when I do it this way `(name, {url: null})` works fine